File: nebras-app/app/src/main/kotlin/com/nebras/mobile/core/config/RemoteBackendConfig.kt
package com.nebras.mobile.core.config

import com.nebras.mobile.BuildConfig

/**
 * التحكم في استدعاءات الـ HTTP القديمة (Render / nebras-backend).
 *
 * - في **التطوير (debug)** لا تُستدعى افتراضيًا حتى لا يعلق التطبيق على timeouts.
 * - في **الإصدار (release)** تبقى مفعّلة كالسابق.
 *
 * بديل `--dart-define=USE_RENDER_BACKEND`: متغيّر بيئة/خاصيّة نظام يُقرأ عند
 * التشغيل، فيمكن تجربة Render أثناء التطوير بلا إعادة بناء:
 * `adb shell setprop debug.nebras.render true`
 */
val isLegacyRenderBackendEnabled: Boolean
    get() {
        val v = runCatching {
            System.getProperty("debug.nebras.render").orEmpty()
        }.getOrDefault("")
        if (v == "true" || v == "1") return true
        if (v == "false" || v == "0") return false
        return !BuildConfig.DEBUG
    }

File: app-flutter/lib/core/config/remote_backend_config.dart
import 'package:flutter/foundation.dart';

/// التحكم في استدعاءات الـ HTTP القديمة (Render / nebras-backend).
///
/// - في **التطوير (debug)** لا تُستدعى افتراضيًا حتى لا تعلق التطبيق على timeouts.
/// - في **الإصدار (release/profile)** تبقى مفعّلة كالسابق.
///
/// لتجربة Render أثناء التطوير:
/// `flutter run --dart-define=USE_RENDER_BACKEND=true`
///
/// لتعطيل Render حتى في الإصدار (نادر):
/// `flutter run --dart-define=USE_RENDER_BACKEND=false`
/// أو عند البناء: `--dart-define=USE_RENDER_BACKEND=false`
bool get isLegacyRenderBackendEnabled {
  const v = String.fromEnvironment('USE_RENDER_BACKEND', defaultValue: '');
  if (v == 'true' || v == '1') return true;
  if (v == 'false' || v == '0') return false;
  return !kDebugMode;
}
